test/common/http/conn_manager_utility_test.cc

```diff
@@ -842,5 +842,119 @@ TEST_F(ConnectionManagerUtilityTest, NonTlsAlwaysForwardClientCert) {
+}
+
+// Client, client enabled, global on.
+TEST_F(ConnectionManagerUtilityTest, ClientSamplingWhenGlobalSet) {
+  EXPECT_CALL(runtime_.snapshot_, featureEnabled("tracing.client_enabled", 100))
+      .WillOnce(Return(true));
+  EXPECT_CALL(runtime_.snapshot_, featureEnabled("tracing.global_enabled", 100, _))
+      .WillOnce(Return(true));
+
+  Http::TestHeaderMapImpl request_headers{
+      {"x-client-trace-id", "f4dca0a9-12c7-4307-8002-969403baf480"},
+      {"x-request-id", "125a4afb-6f55-44ba-ad80-413f09f48a28"}};
+  callMutateRequestHeaders(request_headers, Protocol::Http2);
+
+  EXPECT_EQ(UuidTraceStatus::Client,
+            UuidUtils::isTraceableUuid(request_headers.get_("x-request-id")));
+}
+
+// Client, client disabled, global on.
+TEST_F(ConnectionManagerUtilityTest, NoTraceWhenClientSamplingNotSetAndGlobalSet) {
+  EXPECT_CALL(runtime_.snapshot_, featureEnabled("tracing.client_enabled", 100))
+      .WillOnce(Return(false));
+  EXPECT_CALL(runtime_.snapshot_, featureEnabled("tracing.global_enabled", 100, _))
+      .WillOnce(Return(true));
+
+  Http::TestHeaderMapImpl request_headers{
+      {"x-client-trace-id", "f4dca0a9-12c7-4307-8002-969403baf480"},
+      {"x-request-id", "125a4afb-6f55-44ba-ad80-413f09f48a28"}};
+  callMutateRequestHeaders(request_headers, Protocol::Http2);
+
+  EXPECT_EQ(UuidTraceStatus::NoTrace,
+            UuidUtils::isTraceableUuid(request_headers.get_("x-request-id")));
+}
+
+// Forced, global on.
+TEST_F(ConnectionManagerUtilityTest, ForcedTracedWhenGlobalSet) {
+  ON_CALL(config_, useRemoteAddress()).WillByDefault(Return(false));
+  // Internal request, make traceable.
+  TestHeaderMapImpl headers{{"x-forwarded-for", "10.0.0.1"},
+                            {"x-request-id", "125a4afb-6f55-44ba-ad80-413f09f48a28"},
+                            {"x-envoy-force-trace", "true"}};
+  EXPECT_CALL(random_, uuid()).Times(0);
+  EXPECT_CALL(runtime_.snapshot_, featureEnabled("tracing.global_enabled", 100, _))
+      .WillOnce(Return(true));
+
+  EXPECT_EQ((MutateRequestRet{"10.0.0.1:0", true}),
+            callMutateRequestHeaders(headers, Protocol::Http2));
+  EXPECT_EQ(UuidTraceStatus::Forced, UuidUtils::isTraceableUuid(headers.get_("x-request-id")));
+}
+
+// Forced, global off.
+TEST_F(ConnectionManagerUtilityTest, NoTraceWhenForcedTracedButGlobalNotSet) {
+  ON_CALL(config_, useRemoteAddress()).WillByDefault(Return(false));
+  // Internal request, make traceable.
+  TestHeaderMapImpl headers{{"x-forwarded-for", "10.0.0.1"},
+                            {"x-request-id", "125a4afb-6f55-44ba-ad80-413f09f48a28"},
+                            {"x-envoy-force-trace", "true"}};
+  EXPECT_CALL(random_, uuid()).Times(0);
+  EXPECT_CALL(runtime_.snapshot_, featureEnabled("tracing.global_enabled", 100, _))
+      .WillOnce(Return(false));
+
+  EXPECT_EQ((MutateRequestRet{"10.0.0.1:0", true}),
+            callMutateRequestHeaders(headers, Protocol::Http2));
+}
```

Comment:
Need a test for `isTraceableUuid` - currently would be `NoTrace` depending upon outcome of discussion above. I believe the sampling decision needs to be respected, so even if global returns false, the existing decision should remain the same.